sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -1096,15 +1096,20 @@ object TypeCoercion extends TypeCoercionBase {
       }
     }
 
+    private def isIntervalType(dt: DataType): Boolean = dt match {
+      case _: CalendarIntervalType | _: AnsiIntervalType => true
```

Comment:
Question: there is also `DayTimeIntervalType` and `YearMonthIntervalType` et.c and why they are not included in here?

Reply:
My bad. I was looking at the Object. The class of `DayTimeIntervalType` and `YearMonthIntervalType` has already extended `AnsiIntervalType`

Reply:
> Question: there is also DayTimeIntervalType and YearMonthIntervalType et.c and why they are not included in here?

They are of AnsiIntervalType